src/shared/messaging/port-rpc.ts

```diff
@@ -249,6 +264,25 @@ export class PortRPC<OnMethod extends string, CallMethod extends string>
   connect(port: MessagePort) {
     this._port = port;
     this._listeners.add(port, 'message', event => this._handle(event));
+
+    // For browsers that support a `close` event for MessagePort, we use that
+    // to identify when the other end disconnects. This is translated into a
+    // message event that is similar to what we receive in older browsers
+    // which use a Window unload handler instead.
+    this._listeners.add(port, 'close', event => {
+      port.dispatchEvent(
+        new MessageEvent('message', {
+          data: {
+            protocol: PROTOCOL,
+            version: VERSION,
+            arguments: [],
+            method: 'close',
+            sequence: -1,
+          } satisfies RequestMessage,
```

Comment:
I see the `sendCall` function is internally using a similar `RequestMessage` object. Perhaps we could create a local helper function which creates that kind of object and then we can use it both here and in `sendCall`.

```ts
function requestMessage(
  method: string,
  args: unknown[] = [],
  sequence = -1,
): RequestMessage {
  return {
    protocol: PROTOCOL,
    version: VERSION,
    arguments: args,
    method,
    sequence,
  };
}

// [...]

function sendCall(
  port: MessagePort,
  method: string,
  args: unknown[] = [],
  sequence = -1,
) {
  port.postMessage(requestMessage(
    method,
    args,
    sequence,
  ));
}

// [...]

port.dispatchEvent(
  new MessageEvent('message', {
    data: requestMessage('close')
  }),
);
```